extensions/Lily/MoreEvents.js

```diff
@@ -205,6 +212,33 @@
         runtime.shouldExecuteStopClicked = false;
         originalGreenFlag.call(this);
       };
+
+      let pauseHandler = () => {
+        runtime.allScriptsByOpcodeDo(
+          "lmsMoreEvents_whenProjectPaused",
+          (script, target) => {
+            const topBlockId = script.blockId;
+            for (let j = 0; j < runtime.threads.length; j++) {
+              if (
+                runtime.threads[j].topBlock === topBlockId &&
+                runtime.threads[j].status !== 4
+              ) {
+                return;
+              }
+            }
+            runtime._pushThread(topBlockId, target);
+          }
+        );
+        runtime.removeListener("BEFORE_EXECUTE", pauseHandler);
+      };
+
+      runtime.on("RUNTIME_PAUSED", () => {
+        runtime.on("BEFORE_EXECUTE", pauseHandler);

```

Comment:
You can use .once instead of .on and it will only fire once. Easier to understand than using .on with .removeListener